### ISSUE TITLE
Error 1x127

### ISSUE THREAD

**Issue by rfbarangan** [Ben216k/Patched-Sur#2]
**Describe the bug**
Running Patched Kext on Mac Early 2013 resulted below

**To Reproduce**
1. Update Patched Sur to v1.0.1(122)
2. Update Mac to OS 11.6
3. After reboot run Patched Kext

Error 1x127
Message: "/usr/local/lib/Patched-Sur-Patches/Scripts/NeededPatches.sh: line 74: /usr/local/lib/Patched-Sur-Patches/Scripts/../PatchSystem.sh: No such file or directory"
Output: "Checking environment...
[INFO] We're booted into full macOS.
[INFO] Using dirname source.

Confirming patch location...
Set to detect patches, restarting PatchSystem with NeededPatches...
(2012):BOOTPLIST
(MORE):WIFI
Running PatchSystem.sh..."

**Expected behavior**
After running the patched kext and reboot WIFI and bluetooth should be working. 

**Screenshots**

**Desktop (please complete the following information):**
 - Upgrading from: Big Sur 11.5.2
- Upgrading to: 11.6
 - Mac Model: Early 2013

Work Around:
Edit file in /usr/local/lib/Patched-Sur-Patches/Scripts/
Line +56 of PatchSystem.sh
Line +74 of NeededPatches.sh 
Provide full path of the scripts. 

If there are any red boxes, click on them to copy the logs.

**Comment by ashiddokuru** [comment]
Another work around is using Patch Kext through the post install app from the original Patched Sur v1.0.0 via USB (offload). That's how I got it to work.  Hope this helps too.

**Comment by Ben216k** [comment]
It'll be an in-place fix. Since it's a bug with a script bundled with the patches and the patches are hosted externally (Ben216k/Patched-Sur-Patches) I can just update those then the v1.0.1 patcher will start using the fix.

I'll do a fix sometimes soonish.

**Comment by mezhdutem** [comment]
> Another work around is using Patch Kext through the post install app from the original Patched Sur v1.0.0 via USB (offload). That's how I got it to work. Hope this helps too.
> 
> Also did this work around help you? Can you elaborate a little more on your work around? Thanks!

could you please send the original Patched Sur v1.0.0?

**Comment by mvmagalhaes** [comment]
> > Another work around is using Patch Kext through the post install app from the original Patched Sur v1.0.0 via USB (offload). That's how I got it to work. Hope this helps too.
> > Also did this work around help you? Can you elaborate a little more on your work around? Thanks!
> 
> could you please send the original Patched Sur v1.0.0?
[Patched Sur.zip](https://github.com/Ben216k/Patched-Sur/files/7449862/Patched.Sur.zip)

**Comment by Ben216k** [comment]
I didn't even touch Patch Kexts in v1.0.1, that wouldn't make a difference. The thing that changed it was the fact you had your USB plugged in.

**Comment by Morrile37** [comment]
Still no luck, ran every script on offer and it still fails. Deleted the Patched Sur and rebooted, downloaded another copy still the same. Cannot install Patched Kexts for love or money!

**Comment by Ben216k** [comment]
This has literally no reason not to work. You can tell that's true because of what the commit name is.

```
PSPATH=~/.patched-sur
cd $PSPATH && rm -rf Patched-Sur-Patches*
curl -Lo Patched-Sur-Patches.zip https://codeload.github.com/BenSova/Patched-Sur-Patches/zip/0c10dee
cd /usr/local/lib && sudo rm -rf Patched-Sur-Patches
sudo unzip -q $PSPATH/Patched-Sur-Patches.zip
sudo mv Patched-Sur-Patches-0c10dee Patched-Sur-Patches
```

**Comment by Morrile37** [comment]
I did not realise that the Pen Drive had to be plugged in before running the code, copied files over and now working via your app. Many thanks for your patience.

**Comment by mezhdutem** [comment]
> > Another work around is using Patch Kext through the post install app from the original Patched Sur v1.0.0 via USB (offload). That's how I got it to work. Hope this helps too.
> > Also did this work around help you? Can you elaborate a little more on your work around? Thanks!
> 
> could you please send the original Patched Sur v1.0.0? [Patched Sur.zip](https://github.com/Ben216k/Patched-Sur/files/7449862/Patched.Sur.zip)

Thank you. But even with this one I can't patch mye ketxts. After downloading Patched-Sur-Patches via terminal command I just get the message no patches available

**Comment by mezhdutem** [comment]
> I'm not surprised about the app not working, but did you have Ethernet when you ran the command?

Yes I did. In fact I'm writing now on this particular mac mini late 2012

**Comment by mezhdutem** [comment]
> I'm not surprised about the app not working, but did you have Ethernet when you ran the command?

If there is any possibility to run wifi patch without the app?

**Comment by Ben216k** [comment]
What error is that?

**Comment by mezhdutem** [comment]
> > I'm not surprised about the app not working, but did you have Ethernet when you ran the command?
> 
> If there is any possibility to run wifi patch without the app? (BTW I even can't create an USB installer. While patching installer I get the error



> What error is that?

Message: "  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of /Users/me/.patched-sur/Patched-Sur-Patches.zip or
        /Users/me/.patched-sur/Patched-Sur-Patches.zip.zip, and cannot find /Users/me/.patched-sur/Patched-Sur-Patches.zip.ZIP, period."
Output: "Archive:  /Users/me/.patched-sur/Patched-Sur-Patches.zip"

**Comment by Ben216k** [comment]
That's weird, I have no explanation beyond that. I'll have to look into that one more.

**Comment by Ben216k** [comment]
Regarding the actual error this issue talks about, I have a solution, it'll just take a bit to implement.

**Comment by JonathanSMachado** [comment]
I fixed the 1x127 error by running the sudo ./PatchKexts.sh command, inside the /usr/local/lib/Patched-Sur-Patches/Scripts path.

**Comment by Ben216k** [comment]
Not really a fix, but that is another way to patch the kexts.

**Comment by JonathanSMachado** [comment]
> Not really a fix, but that is another way to patch the kexts.

really....

**Comment by Ben216k** [comment]
PatchKexts.sh and PatchSystem.sh are two different scripts. PatchSystem.sh is the more optimized of the two, so Patched Sur prefers to use that.

**Comment by fariasmark** [comment]
> Yeah that's normal. Try this
> 
> ```
> PSPATH=~/.patched-sur
> cd $PSPATH && rm -rf Patched-Sur-Patches*
> curl -Lo Patched-Sur-Patches.zip https://codeload.github.com/Ben216k/Patched-Sur-Patches/zip/3ee2d76
> cd /usr/local/lib && sudo rm -rf Patched-Sur-Patches || echo "Failed to delete something, it should be fine"
> sudo unzip -q $PSPATH/Patched-Sur-Patches.zip
> sudo mv Patched-Sur-Patches-3ee2d76 Patched-Sur-Patches
> ```

Do that  ☝ and then:
```
cd /usr/local/lib/Patched-Sur-Patches/
sudo cp Scripts/PatchSystem.sh .
sudo cp Scripts/NeededPatches.sh .
sudo chmod 755 PatchSystem.sh
sudo chmod 755 NeededPatches.sh
```

then open Patched Sur.app and Patch Kexts.

This work for me!

**Comment by Morrile37** [comment]
upgraded to 11.6.2 yet cannot patch kexts, same error as before (I believe). I tried to create a USB as before and cannot due to a repetition of the error below.

Error 1x127
Message: "/usr/local/lib/Patched-Sur-Patches/Scripts/PatchSystem.sh: line 66: /usr/local/lib/Patched-Sur-Patches/NeededPatches.sh: No such file or directory"
Output: "Checking environment...
[INFO] We're booted into full macOS.
[INFO] Using usr lib source.

Confirming patch location...
Set to detect patches, restarting PatchSystem with NeededPatches..."

**Comment by Ben216k** [comment]
I have tried (yet again) to fix this stupid bug. Creating a new USB installer, updating macOS or running the following command (you'll need Ethernet for the command) should update the internal patch scripts.

```
PSPATH=~/.patched-sur
cd $PSPATH && rm -rf Patched-Sur-Patches*
curl -Lo Patched-Sur-Patches.zip https://codeload.github.com/Ben216k/Patched-Sur-Patches/zip/3268880
cd /usr/local/lib && sudo rm -rf Patched-Sur-Patches || echo "Failed to delete something, it should be fine"
sudo unzip -q $PSPATH/Patched-Sur-Patches.zip
sudo mv Patched-Sur-Patches-3268880 Patched-Sur-Patches
```

**Comment by steniomelo** [comment]
> I fixed the 1x127 error by running the sudo ./PatchKexts.sh command, inside the /usr/local/lib/Patched-Sur-Patches/Scripts path.

It worked perfeclty for me. Tks @JonathanSMachado

**Comment by PatchedSur4Eva** [comment]
@Ben216k is the MAN! Love the guy! a million thanks! I'm 1000% NOT ungrateful,

but i tried @Ben216k 's command posted Dec 16, 2021 with ethernet, and it didn't work for me.

>>PSPATH=~/.patched-sur
cd $PSPATH && rm -rf Patched-Sur-Patches*
curl -Lo Patched-Sur-Patches.zip https://codeload.github.com/Ben216k/Patched-Sur-Patches/zip/3268880
cd /usr/local/lib && sudo rm -rf Patched-Sur-Patches || echo "Failed to delete something, it should be fine"
sudo unzip -q $PSPATH/Patched-Sur-Patches.zip
sudo mv Patched-Sur-Patches-3268880 Patched-Sur-Patches

 "Advanced Kext" also caused PatchedSur.app to Crash giving me this error

>>Application Specific Information:
Patched_Sur/PatchKextsView.swift:565: Fatal error: Error 1x1
Message: "sudo: /usr/local/lib/Patched-Sur-Patches/Scripts/NeededPatches.sh: command not found"
Output: ""

what DID work was @fariasmark reply


> > Yeah that's normal. Try this
> > ```
> > PSPATH=~/.patched-sur
> > cd $PSPATH && rm -rf Patched-Sur-Patches*
> > curl -Lo Patched-Sur-Patches.zip https://codeload.github.com/Ben216k/Patched-Sur-Patches/zip/3ee2d76
> > cd /usr/local/lib && sudo rm -rf Patched-Sur-Patches || echo "Failed to delete something, it should be fine"
> > sudo unzip -q $PSPATH/Patched-Sur-Patches.zip
> > sudo mv Patched-Sur-Patches-3ee2d76 Patched-Sur-Patches
> > ```
> 
> Do that ☝ and then:
> 
> ```
> cd /usr/local/lib/Patched-Sur-Patches/
> sudo cp Scripts/PatchSystem.sh .
> sudo cp Scripts/NeededPatches.sh .
> sudo chmod 755 PatchSystem.sh
> sudo chmod 755 NeededPatches.sh
> ```
> 
> then open Patched Sur.app and Patch Kexts.
> 
> This work for me!

**Comment by Ben216k** [comment]
Assuming this hasn't gotten a comment in 6 months and I'm 99% sure that I fixed this completely once and for all a while ago, I'm going to close this issue.
